misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/model/JClass.java

```diff
@@ -251,12 +189,21 @@ private void populateFields(Field[] fields) {
     private void populateImplementedInterfaces(Class[] interfaces) {
         for (Class interfaceClass : interfaces) {
             setAllClasses(getAlias(interfaceClass));
-            superClasses.add(getAlias(interfaceClass));
-            superClassNames.add(interfaceClass.getName());
-            if (interfaceClass.getInterfaces() != null) {
-                populateImplementedInterfaces(interfaceClass.getInterfaces());
+        }
+    }
+
+    private boolean isOverloaded(Method method) {
+        boolean overloaded = false;
+        int count = 0;
+        for (Method m : getMethodsAsList(currentClass)) {
+            if (m.getName().equals(method.getName())) {
+                count++;
             }
         }
+        if (count > 1) {
+            overloaded = true;
+        }
+        return overloaded;
```

Comment:
We can get rid of the count variable. Use `.stream.filter()` on getMethodsAsList(currentClass).

https://www.baeldung.com/java-stream-findfirst-vs-findany#using-the-streamfindany . Use the `Optional` for true/false